src/context.rs

```diff
@@ -35,6 +36,8 @@ pub trait ContextTrait {
               ws_port: Option<u16>) -> Shared<Self>;
     fn add_service(&mut self, service: Box<Service>);
     fn remove_service(&mut self, id: String);
+    fn add_socket(&mut self, socket_out: ws::Sender);
+    fn remove_socket(&mut self, id: ws::util::Token);
```

Comment:
I would rather have remove_socket() also take a ws::Sender. No need to show the implementation detail used by add_socket()